x-pack/plugins/observability_solution/profiling/public/services.ts

```diff
@@ -167,5 +179,22 @@ export function getServices(): Services {
       )) as IndicesStorageDetailsAPIResponse;
       return eventsMetricsSizeTimeseries;
     },
+    fetchTopNFunctionAPMTransactions: async ({
+      functionName,
+      http,
+      serviceNames,
+      timeFrom,
+      timeTo,
+    }) => {
+      const query: HttpFetchQuery = {
+        timeFrom,
+        timeTo,
+        functionName,
+        serviceNames: JSON.stringify(serviceNames),
+      };
+      return (await http.get(paths.APMTransactions, {
+        query,
+      })) as Promise<APMTransactionsPerService[]>;
```

Comment:
Could we let the client resolve the promise instead of doing it here?
```suggestion
      return http.get(paths.APMTransactions, {
        query,
      }) as Promise<APMTransactionsPerService[]>;
```